beacon_node/eth1/src/service.rs

```diff
@@ -860,11 +900,30 @@ async fn download_eth1_block(
 mod tests {
     use super::*;
     use toml;
+    use types::MainnetEthSpec;
 
     #[test]
     fn serde_serialize() {
         let serialized =
             toml::to_string(&Config::default()).expect("Should serde encode default config");
         toml::from_str::<Config>(&serialized).expect("Should serde decode default config");
     }
+
+    #[test]
+    fn block_cache_size() {
+        let mut config = Config::default();
+
+        let spec = MainnetEthSpec::default_spec();
+
+        config.set_block_cache_truncation::<MainnetEthSpec>(&spec);
+
+        let len = config.block_cache_truncation.unwrap();
+
+        let seconds_per_voting_period =
+            <MainnetEthSpec as EthSpec>::SlotsPerEth1VotingPeriod::to_u64()
+                * (spec.milliseconds_per_slot / 1000);
+        let eth1_blocks_per_voting_period = seconds_per_voting_period / spec.seconds_per_eth1_block;
+
+        assert!(len > eth1_blocks_per_voting_period as usize * 2);
```

Comment:
I'm not very sure what this test is checking here. `block_cache_truncation` could be less than the old block cache size and the test would still pass.

Reply:
Basically I'm trying to check that with all our fancy math we don't end up with something smaller than the very minimum required. I added some more complexity to it to make it a better test :) Thanks.